Handle all page changes via page store

lonvia opened the pull request:
The `refresh_page()` is now the go-to function for all page-internal linking. It now receives the page and parameters (as URLSearchParams), sets the page store accordingly and also updates the window.location. When called without parameters, refresh_page uses the parameters from the window location to update the page store (exactly like it did until now).

Pages directly subscribe to the page store and take the URL parameters from there. The PR also changes the ReverseBar to use refresh_page() instead of indirectly sending its form. This avoids the annoying double reload when clicking on the map.

mtmail commented:
Wonderful.

* When clicking the branding logo the generated URL becomes `/search.html?[object Object]`
* Same when on /reverse.html clicking 'forward search'
* When on a /details.html result page and clicking 'link to this page', then a page gets loaded. Would it make sense to inspect the current and clicked URL and if they match not reload?
* When on a /details.html result page, scrolling down and clicking a 'details' link then a new place is loaded, but the browser scroll position stays the same. Any reason why the users shouldn't always be taken to the top of the page after a page load? https://developer.mozilla.org/en-US/docs/Web/API/History/scrollRestoration might work.

lonvia commented:
I've fixed the URLs and added a command to scroll the details page to the top on reload.

Not reloading is a bit tricky because it uses a URLSearchParam in the page store where svelte can't detect changes. It likely would work if we reverted to using a simple object for the parameters. Might be worth considering it because in the end the gain of using the class in the page store was minimal.

lonvia commented:
Reverting to simple object parameters doesn't help either and does make the code more verbose. I'm leaning towards just ignoring the additional page load.

mtmail commented:
Different issue now: when on a result page, e.g. /reverse.html?lat=48.39028&lon=11.71576 and clicking the brand logo or 'forward search' (/search.html) then the result list and map are still filled with the results.

lonvia commented:
Last commit now clears the result list when no usable parameters are found to send a request to the API.
